### ISSUE TITLE
C# GetOutput("logits") API

### ISSUE THREAD

**Issue by ambroser53** [microsoft/onnxruntime-genai#1005]
Is there any reason that the C# API wouldn't have the same interface for retrieving the logits that the python API (#956) and C API (#755) have? With the exposing of the adapters API #998 this would be incredibly helpful.

Also #956 introduced the set logits interface although I haven't seen much discussion of it's use cases. I may be wrong but if this allows for the force setting of tokens mid-generation then this would be incredibly useful (for example, generate tokens, check condition, add n tokens, generate more tokens). The current C# API does not currently (unless I am mistaken) allow for the altering of input_ids within the gen_params mid-inference. Is there an alternate method to doing this? Or is the set logits interface the appropriate method?

**Comment by ambroser53** [comment]
Realised the second part of my query is being addressed in "continuous decoding" as referenced in #828 and #967 so the request here is more a c# interface which I see is currently underway in aciddelgado/continuous

**Comment by baijumeswani** [comment]
I think https://github.com/microsoft/onnxruntime-genai/pull/998 adds that API.
Note that the get logits API (in https://github.com/microsoft/onnxruntime-genai/pull/956) is different than the get outputs ("logits") API in that one of them get's you the logits as used by the search and sampling functions and the other gets you the raw model outputs.

**Comment by baijumeswani** [comment]
```GetOutput("logits")``` will return the raw output from the model. It requires that the underlying ONNX model has an output called `logits`
```

The `GetLogits` API is not added to C# yet. But we will add it soon.

I'll mark this issue as completed since we have the `GetOutput("logits")` API now.
